clients/cobol-lsp-vscode-extension/src/commands/RenumCommand.ts

```diff
@@ -0,0 +1,184 @@
+const leftDigits = 6;
+const leftMultiplier = 100;
+
+const rightStart = 72;
+const rightEnd = 80;
+const rightMultiplier = 1000;
+const rightDigits = 8;
+/**
+ * Renumber 1-7 or 73-80 columns in active editor.
+ *
+ * @param actionType RenumberAction
+ */
+export async function renumberLines(
+  actionType: RenumberAction,
+  activeEditor: vscode.TextEditor | undefined = vscode.window.activeTextEditor,
+) {
+  if (!activeEditor) {
+    return;
+  }
+  await notificationHandler(
+    renum,
+    activeEditor,
+    actionType,
+    "Renumbering Sequential Lines...",
+  );
+}
+/**
+ * Remove sequential numbers at 1-7 or 73-80 columns in active editor.
+ *
+ * @param actionType RenumberAction
+ */
+export async function unNumberLines(
+  actionType: RenumberAction,
+  activeEditor: vscode.TextEditor | undefined = vscode.window.activeTextEditor,
+) {
+  if (!activeEditor) {
+    return;
+  }
+
+  await notificationHandler(
+    unNumber,
+    activeEditor,
+    actionType,
+    "Removing Sequential Lines...",
+  );
+}
+
+async function unNumber(
+  textEditor: vscode.TextEditor,
+  actionType: RenumberAction,
+) {
+  const { start, end } =
+    actionType == RenumberAction.LEFT
+      ? {
+          start: leftStart,
+          end: leftEnd,
+        }
+      : {
+          start: rightStart,
+          end: rightEnd,
+        };
+
+  await textEditor.edit((editBuilder) => {
+    for (let i = 0; i < textEditor.document.lineCount; i++) {
+      if (textEditor.document.lineAt(i).text.charAt(start) != "*") {
+        const range = new vscode.Range(
+          new vscode.Position(i, start),
+          new vscode.Position(i, end),
+        );
+        editBuilder.replace(range, " ".repeat(6));
+      }
+    }
+  });
+}
+
+async function renum(
+  textEditor: vscode.TextEditor,
+  actionType: RenumberAction,
+) {
+  const { start, end, multiplier, digits } =
+    actionType == RenumberAction.LEFT
+      ? {
+          start: leftStart,
+          end: leftEnd,
+          multiplier: leftMultiplier,
+          digits: leftDigits,
+        }
+      : {
+          start: rightStart,
+          end: rightEnd,
+          multiplier: rightMultiplier,
+          digits: rightDigits,
+        };
+
+  await textEditor.edit((editBuilder) => {
+    const lineCount = textEditor.document.lineCount;
+    if (lineCount > maxLines) return;
+    for (let i = 0; i < lineCount; i++) {
+      const line = textEditor.document.lineAt(i);
+      const text = line.text;
+      const divider = i > 99999 ? 100 : i > 9999 ? 10 : 1;
+      const value = (i + 1) * (multiplier / divider);
+      const range = new vscode.Range(
+        new vscode.Position(i, start),
+        new vscode.Position(i, end),
+      );
+      if (actionType == RenumberAction.RIGHT && text.length <= start) {
+        const padLength = start - text.length;
+        const placeholderRange = new vscode.Position(i, text.length);
+        editBuilder.replace(placeholderRange, " ".repeat(padLength));
+      }
```

Comment:
I guess something similar can happen on the left side or when unnumbering?

Also perform the action in one edit per line.

Reply:
the case not applies on left due to that replace starts from 0 on left side, but removed since it is trivial to check.
 unnumbering has different logic to prevent redundant changes.